ideagenerator/generator.js

```diff
@@ -5,18 +5,27 @@ let recently_used = [];
 let character_is_group = false;
 let character_post_description = '';
 
-window.onload = function onLoad() {
+function parseCategories() {
 	// Store list of entries by category name
 	for (let i = 0; i < category_names.length; i ++) {
 		name = category_names[i];
 		categories[name] = getCategory(name);
 	}
 }
 
+let parsed = false;
+
 function generate() {
-	reset();
-	let template = pickRandom('template');
-	let result = fillInTemplate(template);
+	if(!parsed)
+		parseCategories();
+	let result;
+	do
+	{
+		reset();
+		let template = pickRandom('template');
+		//Temporary fix for 'result' sometimes being null
+		result = fillInTemplate(template);
+	}while(typeof result === 'undefined' || result.length === 0);
```

Comment:
```suggestion
	} while(typeof result === 'undefined' || result.length === 0);
```

Reply:
I had to turn my auto-formater for this project so there may have been some sections that I may have missed. Thanks!